ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java

```diff
@@ -240,7 +240,7 @@ public Optional<Materialization> getMaterialization(
 
   @Override
   public void stop() {
-    stop(false);
+    stop(true);
```

Comment:
The variable name `isCreateOrReplace` on the various `stop()` methods is really, really confusing.

I'd suggest cleaning that up if possible.

Also, should this have a unit test?

Reply:
Yeah, sorry this isn't the full PR yet. Im not sure if changing the default is actually what we want to do. Defiantly change the shut down action. But with this change you should be able to test

Reply:
We don't want to reset offsets here right?

Reply:
What actually calls this?